src/test/ui/impl-trait/static-return-lifetime-infered.nll.stderr

```diff
@@ -17,6 +17,10 @@ LL |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> {
    |                         - let's call the lifetime of this reference `'1`
 LL |         self.x.iter().map(|a| a.0)
    |         ^^^^^^^^^^^^^ requires that `'1` must outlive `'static`
+help: you can add a constraint to the return type to make it last less than `'static` and match '1
+   |
+LL |     fn iter_values_anon(&self) -> impl Iterator<Item=u32> + '1 {
+   |                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
This, on the other hand, is very incorrect. This is handled on the current suggestion by proposing `'_` instead.

Reply:
Pushed a fix for this.